### PR TITLE
auto select default channel

### LINKED ISSUE #850: display note if no laps are selected.
If no laps are selected prompt user to select a lap. Display a message in the analysis widget(s) area.

### PR DESCRIPTION
**Pull request by brentpicasso**
* Automatically select a default channel if no channels are already selected.
* Display a helpful message in the line chart to prompt the user to select a lap or channel, as appropriate.

Please review after https://github.com/autosportlabs/RaceCapture_App/pull/840 is reviewed, as I had to merge that branch into this one to verify certain functionality.

Resolves #850 and #824

### CODE REVIEW

File: autosportlabs/racecapture/views/analysis/analysiswidget.py [hunk 1]
```diff
@@ -75,20 +76,32 @@ def on_options_enabled(self, instance, value):
...
+        Remove a lap specified by the source reference
+        :param source_ref indicating the selected session / lap
+        :type SourceRef
+        '''
         try:
-            self.on_lap_removed(lap_ref)
-            del(self.selected_laps[str(lap_ref)])
+            self.on_lap_removed(source_ref)
+            del(self.selected_laps[str(source_ref)])
+            self.laps_selected = bool(self.selected_laps)
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/analysis/analysiswidget.py]: So `selected_laps` is a hash and we're casting it to a bool... is there a more clear way to do what we mean? I'm guessing we really want to know if there are any selected laps, correct? Can you get a length of a hash?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/analysis/analysiswidget.py]: From research, apparently that is the pythonic way to determine if a dictionary is empty!

Comment (ryandoherty) [inline autosportlabs/racecapture/views/analysis/analysiswidget.py]: ![]
(http://www.reactiongifs.com/wp-content/uploads/2013/06/oh-really-now.gif)

Comment (brentpicasso) [inline autosportlabs/racecapture/views/analysis/analysiswidget.py]: Yarr, pretty cool. "falsey" values :) http://stackoverflow.com/questions/12997305/empty-list-boolean-value

**Comment by ryandoherty** [comment]
Looks good, works, :shipit:

**Pull request merged**